src/raster/r.hand/r.hand.py

```diff
@@ -0,0 +1,427 @@
+# % description: Inundation depth (single output)
+# % guisection: Parameters
+# %end
+
+# %option
+# % key: start_water_level
+# % type: double
+# % required: no
+# % description: Start water level for flooding simulation
+# % guisection: Parameters
+# %end
+
+# %option
+# % key: end_water_level
+# % type: double
+# % required: no
+# % description: End water level for flooding simulation
+# % guisection: Parameters
+# %end
+
+# %option
+# % key: water_level_step
+# % type: double
+# % required: no
+# % description: Step increment for water level in flooding simulation
+# % answer: 1
+# % guisection: Parameters
+# %end
+
+# %option G_OPT_MEMORYMB
+# %end
+
+# %flag
+# % key: t
+# % description: Generate inundation raster maps for a series of water levels
+# % guisection: Output
+# %end
+
+# %flag
+# % key: m
+# % description: Use memory swap (operation is slow)
+# %end
+
+import sys
+import atexit
+import uuid
+import grass.script as gs
+from grass.exceptions import CalledModuleError
+
+tmp_raster_list = []
+
+
+def cleanup():
+    """ "Remove temporary raster maps"""
+    if len(tmp_raster_list) > 0:
+        gs.run_command(
+            "g.remove",
+            type="raster",
+            name=",".join(tmp_raster_list),
+            # pattern="tmp_*",
+            flags="f",
+            quiet=True,
+        )
+
+
+def run_r_watershed(
+    elevation: str,
+    streams: str,
+    direction: str,
+    threshold: int,
+    memory: int,
+    swap_mode_flag: str,
+) -> None:
+    """
+    Run the GRASS GIS r.watershed module to generate streams and flow direction raster maps.
+
+    Parameters:
+    elevation (str): Name of the input elevation raster map.
+    streams (str): Name of the output streams raster map.
+    direction (str): Name of the output flow direction raster map.
+    threshold (int): Minimum size of exterior watershed basin in cells. Must be greater than 0.
+    memory (int): Maximum memory to be used in MB.
+    swap_mode_flag (str): If provided, enables memory swap mode.
+
+    Returns:
+    None
+    """
+
+    if streams.startswith("tmp") and direction.startswith("tmp"):
+        gs.message(_("Generating streams and flow direction raster maps"))
+        if threshold <= 0:
+            gs.fatal(_("Threshold must be greater than 0"))
+
+        try:
+            gs.run_command(
+                "r.watershed",
+                elevation=elevation,
+                threshold=threshold,
+                stream=streams,
+                drainage=direction,
+                memory=memory,
+                flags=swap_mode_flag,
+                quiet=True,
+            )
+        except CalledModuleError as e:
+            gs.fatal(
+                _("Error generating streams and flow direction raster maps: %s")
+                % e.stderr
+            )
+
+
+def set_hand_colors(hand: str) -> None:
+    """Set HAND raster colors based on Norbre et al. 2011"""
+    hand_colors = """
+        0% white
+        1-5 #1d91c0
```

Comment:
This doesn't look like correct syntax with the dash.

Reply:
I changed it to use the correct syntax.